public/app/features/alerting/unified/components/rule-editor/GrafanaEvaluationBehavior.tsx

```diff
@@ -15,17 +16,24 @@ import { RuleEditorSection } from './RuleEditorSection';
 
 const MIN_TIME_RANGE_STEP_S = 10; // 10 seconds
 
+function parseDurationToMilliseconds(duration: string) {
+  return durationToMilliseconds(parseDuration(duration));
+}
+
 const forValidationOptions = (evaluateEvery: string): RegisterOptions => ({
   required: {
     value: true,
     message: 'Required.',
   },
   pattern: durationValidationPattern,
   validate: (value) => {
-    const evaluateEveryDuration = parseDuration(evaluateEvery);
-    const forDuration = parseDuration(value);
-    const millisFor = durationToMilliseconds(forDuration);
-    const millisEvery = durationToMilliseconds(evaluateEveryDuration);
+    // 0 is a special value meaning for equals evaluation interval
+    if (value === '0' || value === '0s') {
```

Comment:
I'm wondering if `0m` etc. are also accepted options.

Reply:
They should be https://github.com/grafana/grafana/issues/52491

Reply:
Should we use a regular expression instead like `0([mnuµ]?s)|m|h`?

Reply:
Ah, sorry I misread your first comment. I'm checking the other zeros :)

Reply:
The backend accepts `0m` and `0h` but it always sets `0s` in that case (same as just for plain 0). Should we allow them?

Reply:
I'll allow them. If it's something we can parse as a `0` then it's ok (there is an additional pattern validation already)